nservicebus/handlers/accessing-data.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Accessing and modifying data
+summary: How to access business data in sync with message consumption and modifications to NServiceBus-controlled data.
+component: Core
+versions: '[5,)'
+related:
+- nservicebus/nhibernate/accessing-data
+- nservicebus/ravendb
+tags:
+ - NHibernate
+ - RavenDB
+ - Persistence
+ - Data access
+ - Saga
+---
+
+In most cases [handlers](/nservicebus/handlers/) are meant to modify the internal state of an application based on the received message. There are multiple ways to do this.
+
+
+## Without using NServiceBus persistence
+
+The simplest way to modify the state of the application is to execute the data access code in the handler without using NServiceBus persistence features. Transport transaction mode has to be taken into account when designing such code.
+
+
+### Transport using native transactions
+
+When the selected transport is configured to use native transactions, either in `ReceiveOnly` or `SendsAtomicWithReceive` mode, the data access code in the handler can be executed multiple times for a single message. This can lead to data corruption if that code is not *idempotent* (which is a fancy word for ensuring side effects of the code are the same, no matter how many times the code is invoked). Here's an example of data access code that is not idempotent:
```

Comment:
I added the link and slightly change the desctiption but I don't like "safely receive the same message" because it blurs the definition.